apps/server/src/controllers/paymentsController/fundWallet.ts

```diff
@@ -57,39 +57,38 @@ export const fundWallet = async (req: RequestExt, res: Response) => {
         copyBalance = userWallet.balance + _amount
       }
 
-      const transaction = await userWallet.update(
+      const updatedWallet = await userWallet.update(
         { balance: copyBalance },
         { returning: true }
       )
 
       await createTransaction({
-        senderId: _userId,
-        receiverId: transaction.id,
+        senderWalletId: _userId,
+        walletId: updatedWallet.id,
         action: transactionActionType[_transactionType],
-        previousBalance,
-        balance: transaction.balance,
         amount: _amount,
         status: transactionStatusType.PENDING,
         transferType: transactionTransferType.INTERNAL_TRANSFERS,
-        walletType: _walletType,
+        walletType: transactionWalletType.GLOBAL,
       })
 
       return HTTP_STATUS_HELPER[HTTP_STATUS_CODE.SUCCESS](res)
     }
 
     return HTTP_STATUS_HELPER[HTTP_STATUS_CODE.NOT_FOUND](res)
   } catch (error) {
-    // reverse the error if any error occurs
+    // reverse the transaction if any error occurs
+    await Transactions.destroy({ where: { walletId: _userId } })
```

Comment:
i don't think this fully satisfies a reversed transaction in this case

Reply:
Yes. I agree. For a transaction resource to be destroyed, it doesn't mean that the transaction is reversed. Some logic should be employed in this case.

Reply:
Resolved. 
- Added db update operations and transactions creation to a db transaction.
- The transaction is committed if everything goes well.
-  If any error occurs, the transaction is rolled-back to revert any changes